cjdns-bytes/src/utils.rs

```diff
@@ -41,6 +41,14 @@ impl<'a> Reader<'a> {
         Ok(bytes_32_array)
     }
 
+    pub fn read_all_pure(&self) -> &[u8] {
+        self.0
+    }
+
+    pub fn read_all_mut(&mut self) -> &[u8] {
+        self.take_bytes(self.len()).expect("attempting to read data more than slice have")
+    }
+
```

Comment:
Not a good name choices, looking at the methods it is not obvious what do they do and what is the difference.

Reply:
I suggest `read_all_mut` ->  `read_all`. Because it's seen from context, that reader is consumed, so user will get that `read_all` "consumes" underlying type. What's bad about `read_all_pure`? I think `pure` tells enough about reader "consuming".